bundles/org.openhab.binding.epsonprojector/src/main/java/org/openhab/binding/epsonprojector/internal/EpsonProjectorHandler.java

```diff
@@ -0,0 +1,451 @@
+    private @Nullable ScheduledFuture<?> pollingJob;
+
+    public EpsonProjectorHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        String channelId = channelUID.getId();
+        if (command instanceof RefreshType) {
+            Channel channel = this.thing.getChannel(channelUID);
+            if (channel != null) {
+                updateChannelState(channel);
+            }
+        } else {
+            EpsonProjectorCommandType epsonCommand = EpsonProjectorCommandType.getCommandType(channelId);
+            sendDataToDevice(epsonCommand, command);
+        }
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(EpsonProjectorConfiguration.class);
+
+        updateStatus(ThingStatus.UNKNOWN);
+
+        scheduler.execute(() -> {
+            if (StringUtils.isNotEmpty(config.serialPort)) {
+                device = new EpsonProjectorDevice(config.serialPort);
+            } else if (StringUtils.isNotEmpty(config.host) && config.port > 0) {
+                device = new EpsonProjectorDevice(config.host, config.port);
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR);
+                return;
+            }
+
+            try {
+                device.connect();
+                updateStatus(ThingStatus.ONLINE);
+
+                int pollingInterval = config.pollingInterval;
+                if (pollingInterval == 0) {
+                    pollingInterval = 10000;
+                }
+
+                List<Channel> channels = this.thing.getChannels();
+
+                pollingJob = scheduler.scheduleAtFixedRate(new Runnable() {
+                    @Override
+                    public void run() {
+                        for (Channel channel : channels) {
+                            updateChannelState(channel);
+                        }
+                    }
+                }, 0, pollingInterval, TimeUnit.MILLISECONDS);
+
+            } catch (EpsonProjectorException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+            }
+        });
+    }
+
+    @Override
+    public void dispose() {
+        if (pollingJob != null) {
+            pollingJob.cancel(true);
+        }
+        closeConnection();
+        super.dispose();
+    }
+
+    private void updateChannelState(Channel channel) {
+        try {
+            if (!isLinked(channel.getUID())) {
+                return;
+            }
+
+            EpsonProjectorCommandType epsonCommand = EpsonProjectorCommandType.getCommandType(channel.getUID().getId());
+
+            State state = queryDataFromDevice(epsonCommand);
+
+            updateState(channel.getUID(), state);
+        } catch (IllegalArgumentException e) {
+            logger.warn("Illegal command {}", channel.getUID().getId());
```

Comment:
There is no illegal command? There is maybe an unexpected channel.

Reply:
This message supposedly referred to the Epson commands, not openHAB commands, I believe, I changed it to "Unknown channel" since channels controls the equivalent Epson command of the same name - this is clearer.